src/consumer/__tests__/runner.spec.js

```diff
@@ -34,6 +42,37 @@ describe('Consumer > Runner', () => {
     })
   })
 
+  describe('when eachBatchAutoResolve is set to false', () => {
+    let eachBatch
+
+    beforeEach(() => {
+      eachBatch = jest.fn()
+      runner = new Runner({
+        consumerGroup,
+        eachBatchAutoResolve: false,
+        eachBatch,
+        onCrash,
+        logger: newLogger(),
+      })
+      runner.scheduleFetch = jest.fn(() => runner.fetch())
+    })
+
+    it('does not call resolveOffset with the last offset', async () => {
+      const topic = 'topic-name'
+      const partition = 0
+      const batch = new Batch(topic, {
+        partition,
+        highWatermark: 5,
+        messages: [{ offset: 4, key: '1', value: '2' }],
+      })
+
+      consumerGroup.fetch.mockImplementationOnce(() => [batch])
+      await runner.start()
+      expect(onCrash).not.toHaveBeenCalled()
+      expect(consumerGroup.resolveOffset).not.toHaveBeenCalledWith({ topic, partition, offset: 4 })
```

Comment:
Just check that it hasn't been called at all, since we're not testing offset calculation here, and don't want this to accidentally pass.